app/controllers/settings_controller.rb

```diff
@@ -256,6 +256,25 @@ def twitter_disconnect
     return redirect_to "/settings"
   end
 
+  def keybase_auth
+    if !params[:kb_username].present? || !params[:kb_signature].present? || !params[:kb_ua].present? || (!params[:username].present? || !params[:username] == @user.username)
```

Comment:
Can Rails's strong parameters clean this up? You may need to make like a non-db-backed model object named `KeybaseAuth` or something for the params to live on... and that would have the nice effect of pulling much of this complexity out of the controller.

Reply:
I'll be honest I have no idea what any of this means, can you elaborate?

Reply:
It's the Rails security feature for saying "I expect this form to give all these fields, and only these fields": https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters

Reply:
for any users with multiple lobsters accounts, it might be nice to have the `!params[:username] == @user.username` error case handled a little differently. i.e. tell them that they're logged in as the wrong user. otherwise it could be pretty confusing what's actually going wrong.

Reply:
Users are not permitted to have more than one account, it's considered sockpuppeting and leads to banning all but one account, at the least.

I don't try to actively seek it out because I'm ok with (hypothetical scenario here) an Apple employee who wants to gripe about how they don't like Swift without getting in trouble at work, but it's a regular problem with self-promoters who vote up their own links and comments.